### PR TITLE
Security/fix init.d quoting

### PR DESCRIPTION
**Pull request by RichiH**

### CODE REVIEW

**Comment by torkelo** [comment]
this looks great! thank you! 

Trying to google why `${VARNAME}` is preferable, nothing relating to security. 

Only that it allows for interpolation like `"${foo}bar"` .  

My impression about the securty issue was that daemon command line options should be quoted, not seeing that change.

**Comment by RichiH** [comment]
This is approaching a state it can be merged in. Please be advised that I didn't test the script as I don't use init.d any more.

For safety, I am summoning @mirabilos to review what I did.

File: packaging/deb/init.d/grafana-server [hunk 1]
```diff
@@ -22,20 +22,20 @@
 PATH=/bin:/usr/bin:/sbin:/usr/sbin
 NAME=grafana-server
 DESC="Grafana Server"
-DEFAULT=/etc/default/$NAME
+DEFAULT=/etc/default/${NAME}
```
Comment (mirabilos) [inline packaging/deb/init.d/grafana-server]: useless

File: packaging/deb/init.d/grafana-server [hunk 2]
```diff
@@ -22,20 +22,20 @@
...
 
 GRAFANA_USER=grafana
 GRAFANA_GROUP=grafana
 GRAFANA_HOME=/usr/share/grafana
 CONF_DIR=/etc/grafana
-WORK_DIR=$GRAFANA_HOME
+WORK_DIR=${GRAFANA_HOME}
```
Comment (mirabilos) [inline packaging/deb/init.d/grafana-server]: useless

File: packaging/deb/init.d/grafana-server [hunk 3]
```diff
@@ -22,20 +22,20 @@
...
 DATA_DIR=/var/lib/grafana
 PLUGINS_DIR=/var/lib/grafana/plugins
 LOG_DIR=/var/log/grafana
-CONF_FILE=$CONF_DIR/grafana.ini
+CONF_FILE=${CONF_DIR}/grafana.ini
```
Comment (mirabilos) [inline packaging/deb/init.d/grafana-server]: useless

File: packaging/deb/init.d/grafana-server [hunk 4]
```diff
@@ -22,20 +22,20 @@
...
 MAX_OPEN_FILES=10000
-PID_FILE=/var/run/$NAME.pid
-DAEMON=/usr/sbin/$NAME
+PID_FILE=/var/run/${NAME}.pid
+DAEMON=/usr/sbin/${NAME}
```
Comment (mirabilos) [inline packaging/deb/init.d/grafana-server]: both useless

File: packaging/deb/init.d/grafana-server [hunk 5]
```diff
@@ -45,7 +45,7 @@ if [ `id -u` -ne 0 ]; then
 	exit 4
 fi
 
-if [ ! -x $DAEMON ]; then
+if [ ! -x ${DAEMON} ]; then
```
Comment (mirabilos) [inline packaging/deb/init.d/grafana-server]: useless **and** does not quote, i.e. insecure

File: packaging/deb/init.d/grafana-server [hunk 6]
```diff
@@ -57,85 +57,85 @@ if [ -r /etc/default/rcS ]; then
 fi
 
 # overwrite settings from default file
-if [ -f "$DEFAULT" ]; then
-	. "$DEFAULT"
+if [ -f "${DEFAULT}" ]; then
+	. "${DEFAULT}"
 fi
```
Comment (mirabilos) [inline packaging/deb/init.d/grafana-server]: useless change (was secure before)

File: packaging/deb/init.d/grafana-server [hunk 7]
```diff
@@ -57,85 +57,85 @@ if [ -r /etc/default/rcS ]; then
...
 
 DAEMON_OPTS="--pidfile=${PID_FILE} --config=${CONF_FILE} cfg:default.paths.data=${DATA_DIR} cfg:default.paths.logs=${LOG_DIR} cfg:default.paths.plugins=${PLUGINS_DIR}"
 
 case "$1" in
   start)
 
-	log_daemon_msg "Starting $DESC"
+	log_daemon_msg "Starting ${DESC}"
```
Comment (mirabilos) [inline packaging/deb/init.d/grafana-server]: useless

File: packaging/deb/init.d/grafana-server [hunk 8]
```diff
@@ -57,85 +57,85 @@ if [ -r /etc/default/rcS ]; then
...
 
-	pid=`pidofproc -p $PID_FILE grafana`
-	if [ -n "$pid" ] ; then
+	pid=`pidofproc -p ${PID_FILE} grafana`
```
Comment (mirabilos) [inline packaging/deb/init.d/grafana-server]: useless; also, as long as accent gravis-style command substitution is used, this cannot ever be secured; something like…

    pid=$(pidofproc -p "$PID_FILE" grafana)

… would work.

File: packaging/deb/init.d/grafana-server [hunk 9]
```diff
@@ -57,85 +57,85 @@ if [ -r /etc/default/rcS ]; then
...
+	if [ -n "${pid}" ] ; then
 		log_begin_msg "Already running."
 		log_end_msg 0
 		exit 0
 	fi
 
 	# Prepare environment
-	mkdir -p "$LOG_DIR" "$DATA_DIR" && chown "$GRAFANA_USER":"$GRAFANA_GROUP" "$LOG_DIR" "$DATA_DIR"
-	touch "$PID_FILE" && chown "$GRAFANA_USER":"$GRAFANA_GROUP" "$PID_FILE"
+	mkdir -p "${LOG_DIR}" "${DATA_DIR}" && chown "${GRAFANA_USER}":"${GRAFANA_GROUP}" "${LOG_DIR}" "${DATA_DIR}"
```
Comment (mirabilos) [inline packaging/deb/init.d/grafana-server]: useless *and* makes the whole thing *much* harder to read

File: packaging/deb/init.d/grafana-server [hunk 10]
```diff
@@ -1,5 +1,9 @@
 #! /usr/bin/env bash
 
+set -u
```
Comment (mirabilos) [inline packaging/deb/init.d/grafana-server]: OUCH! **WHY?**

**Review by mirabilos — CHANGES_REQUESTED** [review]
after some comments, I stopped… this is going nowhere

**Comment by RichiH** [comment]
@mirabilos I think you reviewed partial commits and not the final state.

**Comment by torkelo** [comment]
Since you did not test it I am unsure I can merge this :(

**Comment by torkelo** [comment]
closing this as contains so many changes that feels unrelated to security and tricky to test all changes

**Pull request closed without merging**